sdk/search/Azure.Search.Documents/src/Spatial/SpatialFormatter.cs

```diff
@@ -0,0 +1,191 @@
+        /// <summary>
+        /// Encodes a polygon for use in OData filters.
+        /// </summary>
+        /// <param name="line">The <see cref="GeometryLineStringProxy"/> to encode.</param>
+        /// <returns>The OData filter-encoded POLYGON string.</returns>
+        /// <exception cref="ArgumentException">The <paramref name="line"/> has fewer than 4 points, or the first and last points do not match.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="line"/> or <see cref="GeometryLineStringProxy.Points"/> is null.</exception>
+        public static string EncodePolygon(GeometryLineStringProxy line)
+        {
+            Argument.AssertNotNull(line, nameof(line));
+            Argument.AssertNotNull(line.Points, $"{nameof(line)}.{nameof(line.Points)}");
+
+            if (line.Points.Count < 4)
+            {
+                throw new ArgumentException(
+                    $"A GeometryLineString must have at least four Points to form a searchable polygon.",
+                    $"{nameof(line)}.{nameof(line.Points)}");
+            }
+            else if (line.Points[0] != line.Points[line.Points.Count - 1])
+            {
+                throw new ArgumentException(
+                    $"A GeometryLineString must have matching first and last Points to form a searchable polygon.",
+                    $"{nameof(line)}.{nameof(line.Points)}");
+            }
+
+            StringBuilder odata = new StringBuilder("geography'POLYGON((");
+
+            bool first = true;
+            foreach (GeometryPointProxy point in line.Points)
+            {
+                if (!first)
+                {
+                    odata.Append(",");
+                }
+                else
+                {
+                    first = false;
+                }
+
+                odata.Append(JsonSerialization.Double(point.Y, CultureInfo.InvariantCulture))
+                    .Append(" ")
+                    .Append(JsonSerialization.Double(point.X, CultureInfo.InvariantCulture));
+            }
+
+            return odata
+                .Append("))'")
+                .ToString();
+        }
+
+        /// <summary>
+        /// Encodes a polygon for use in OData filters.
+        /// <seealso cref="EncodePolygon(GeometryLineStringProxy)"/>
+        /// </summary>
+        /// <param name="polygon">The <see cref="GeometryPolygonProxy"/> to encode.</param>
+        /// <returns>The OData filter-encoded POLYGON string.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="polygon"/> or <see cref="GeometryPolygonProxy.Rings"/> is null.</exception>
+        public static string EncodePolygon(GeometryPolygonProxy polygon)
+        {
+            Argument.AssertNotNull(polygon, nameof(polygon));
+            Argument.AssertNotNull(polygon.Rings, $"{nameof(polygon)}.{nameof(polygon.Rings)}");
+
+            if (polygon.Rings.Count != 1)
+            {
+                throw new ArgumentException(
```

Comment:
Is this a premature validation? Should we let the service do it?

Reply:
Perhaps, but I'm just copying what was already there, which was copied from track 1. @tg-msft this does seem more like client-side validation, so maybe we shouldn't do it? There's a few validations we have to do to avoid `NullReferenceExceptions` and the like, but here only `> 0` really matters.

Reply:
Not really.  This is us doing magic to allow polygons with one ring to be treated like a line string.  There's no format we could serialize a general polygon into that the service could then complain about it.  It's either "explain our magic with a helpful error" or "remove the magic."  I guess I could be talked into either since this hasn't GA'ed yet.

Reply:
I see. That makes sense about teasing out only a single line, and I think keeping this makes sense. @pakrym, in the interest of time since we're releasing a *preview* next week, if you feel strongly about not doing this client-side to only tease out a singe polygon (line), please file a separate issue.